### ISSUE TITLE
B205mini... should it work?

### ISSUE THREAD

**Issue by john-g8cqx** [kb1vc/SoDaRadio#6]
Hi, came across this and thought it looked very good and wanted to try it with an Ettus B205 mini I have on evaluation here. I downloaded source and compiled on my Ubuntu 18.04 laptop. I Have Gnuradio and GQRX already working from source on this machine with the B205.

It finished cmake, compiled and installed OK. I can uhd_find_devices and uhd_usrp_probe, the latter loads the firmware the first time run from the normal gnuradio install ddirectory (/home/g8cqx/prefix/normal/share/uhd/images/)
I've tried setting this in an environment variable without any luck. This is what I get:

g8cqx@g8cqx-Satellite-A300:~/SoDaRadio/build$ echo $UHD_IMAGES_DIR
/home/g8cqx/prefix/normal/share/uhd/images/
g8cqx@g8cqx-Satellite-A300:~/SoDaRadio/build$ SoDaRadio
[INFO] [UHD] linux; GNU C++ version 7.3.0; Boost_106501; UHD_3.14.0.0-release
[WARNING] [B200] EnvironmentError: IOError: Could not find path for image: usrp_b200_fw.hex

Using images directory: <no images directory located>

Set the environment variable 'UHD_IMAGES_DIR' appropriately or follow the below instructions to download the images package.

Please run:

 "/usr/lib/uhd/utils/uhd_images_downloader.py"
terminate called after throwing an instance of 'uhd::key_error'
  what():  LookupError: KeyError: No devices found for ----->
Empty Device Address

Any ideas?
73s John G8CQX

**Comment by dkozel** [comment]
Hi @john-g8cqx,

Do you have UHD installed from source or from the Ubuntu repository?

`/usr/lib/uhd/utils/uhd_images_downloader.py` is showing a system install while it sounds like you're expecting a source install done into your home directory prefix. Can you check out, using `which -a uhd_usrp_probe`, if you have two installs? If you do, remove one (probably the repository one) and try again.

**Comment by john-g8cqx** [comment]
Hi Derek,
Thanks for the prompt and helpful reply. I did have a second UHD install which I deleted leaving only my source build Gnuradio one (I used Pybombs to install that). I have the environment variable pointing to the images set up in .profile so now I booted, ran my normal scrip which takes me to the gnu install directory, sets the gnuradio environment variables then runs uhd_usrp_probe whuich installs the firmware file. Then I type SoDaRadio and hey-presto... it works! First look is very good. It is pretty efficient compared to gnuradio-companion and GQRX which both suffer processing underruns on the old intel centrino laptop (2 core 2G).
I found one immediate issue. On SSB TX I get a 'howl round' because of a high sidetone which I have not been able to mute...I set the sidetone slider on the settings tab to zero. Plugging in earphones to mute the speaker  allowed me to check the TX was OK otherwise. Is there a control somewhere I haven't found?
regards, John G8CQX

**Comment by kb1vc** [comment]
Thanks for trying soda radio

As for the feedback.     

Ooops.  That’s what I get for not testing everything after adding a feature.   

The problem is that I added full duplex operation for the new geosynchronous satellite.    

I won’t be able to fix this until next week.  In the meantime you may find a previous version gives less trouble. Alas I can’t get to the repo to give you a good commit number but it will be a version of the master branch prior to the satellite update that added full duplex

**Comment by kb1vc** [comment]
John, 

If you are using V 6.6.0, then there will be a button in the "Config. Band" tab labeled
"Full Duplex (sat. mode)" that should be turned off by default.  If you still have trouble with audio feedback, let me know, and I'll do some testing.  (The setup on my bench right now doesn't have a microphone, as its tx audio comes from WSJTX. )

matt

**Comment by john-g8cqx** [comment]
Matt, back in the "shack " now. I have got 6.6.0. Just to be sure I did a git pull and I am up to date apparently.

I looked at the config tab as you suggested and the "full duplex" option was not ticked. So I ticked it and un ticked it and the rx feedback had disappeared. Could it be that the default setting is not activated? Anyway, checking and unchecking is an adequate work-around.

Thanks, John.
